src/mocks.ts

```diff
@@ -163,6 +177,20 @@ export function getUid2Cookie() {
   }
 }
 
+export function removeUid2LocalStorage() {
+  localStorage.removeItem(localStorageKeyName);
+}
+
+export function setUid2LocalStorage(identity: any) {
+  const value = JSON.stringify(identity);
+  localStorage.setItem(localStorageKeyName, value);
+}
+
+export function getUid2LocalStorage() {
+  const value = localStorage.getItem(localStorageKeyName);
+  return value !== null ? JSON.parse(value) : undefined;
```

Comment:
I see what you mean, this seems like a bit of copy-paste work and would still mean tests would pass if the actual functionality broke (BAD). Are you suggesting that this mocks file should have a 
```
const uid2LocalStorageManager = new UID2LocalStorageManager();
```

and then be 
```
export function getUid2LocalStorage() {
  return uid2LocalStorageManager.loadIdentityFromLocalStorage();
}
```

Reply:
I wonder if we need them in the mocks.ts? 
Could we do `const uid2LocalStorageManager = new UID2LocalStorageManager(); ` in test file and just call uid2LocalStorageManager functions?

Reply:
yeah I think that makes sense. In fact I think I can actually use 
```
const uid2StorageManager = new UID2StorageManager({});
```
then just do 
```
uid2StorageManager.removeValues();
```

rather than having 
```
  removeUid2Cookie();
  removeUid2LocalStorage();
```

Reply:
I've made some changes, though I still have `getUid2LocalStorage()` as a mock so that it will return `undefined` instead of `null`. Alternatively, I could change `loadIdentityFromLocalStorage()` to return `Uid2Identity | undefined` instead of `Uid2Identity | null` and then remove the need for `getUid2LocalStorage()`

I initially based it off `loadIdentityFromCookie(): Uid2Identity | null` but I think maybe we can use `undefined` instead of `null`. What do you think?

Reply:
Sorry, I don't think I get this, what is the difference between return `undefined ` and `null` here? why we want it return null in SDK but undefined in tests?

Reply:
tbh I am getting a bit confused myself. - perhaps @lionell-pack-ttd could chime in next week. I see in [getUid2Cookie()](https://github.com/IABTechLab/uid2-web-integrations/blob/main/src/mocks.ts#L154C5-L154C5) we can return `undefined` (so this is what the tests expect), whereas in [loadIdentityFromCookie()](https://github.com/IABTechLab/uid2-web-integrations/blob/f0a852748bf97143d874ee3c1458044f06705183/src/uid2CookieManager.ts#L74) we can return `null`. Would it make sense to just use `null` or `undefined` everywhere, across SDK, tests, cookies & local storage?

Reply:
What are we achieving by switching from the original approach to `loadIdentityFromLocalStorage`? Test infrastructure shouldn't rely on the things it's testing. If the goal is to test whether a value is set in local storage (or clear the value from local storage), then using the UID2LocalStorageManager to do that is a problem.

For example, with this approach, if I change UID2LocalStorageManager to this:
![image](https://github.com/IABTechLab/uid2-web-integrations/assets/111951647/b98884f2-7ea6-4f43-bcea-55a1c701d48a)

The tests all still pass, even though UID2LocalStorageManager doesn't use localStorage at all and would be horribly broken in a real context (it would never persist any data and always start as null on each page load).

Reply:
thanks for the feedback Lionell, I've reverted the change so that I'm using `getUid2LocalStorage` again. One thing I wanted to check with you again @lionell-pack-ttd - are you okay with the implementation of `getUid2LocalStorage`? 

Specifically, the ` : undefined` is necessary because the mock method `getUid2Cookie` returns `undefined`, and we have tests that run for both local storage and cookie that assert on `toBeUndefined()`. 

Other alternatives includes:
* Changing `getUid2Cookie` to return `null` if `!docCookie`, then change `getUid2LocalStorage` to return `null` instead of `undefined`
* Changing the assertions to be `toBeFalsy()` instead of `toBeUndefined()`. The issue with this is that it also includes `false`, `0` and `''` which I don't particularly like - as far as I can tell that is the only way to cover `toBeNull() || toBeUndefined()`

I'd lean towards the first alternative unless you feel strongly about it.

Reply:
The first approach seems like a good one to me - but I don't really feel strongly here, whichever you think is best.

Reply:
I've gone with this approach:
> Changing getUid2Cookie to return null if !docCookie, then change getUid2LocalStorage to return null instead of undefined